crates/ty_python_semantic/src/types/function.rs

```diff
@@ -905,6 +940,12 @@ impl<'db> FunctionType<'db> {
         self.literal(db).iter_overloads_and_implementation(db)
     }
 
+    pub(crate) fn first_overload(self, db: &'db dyn Db) -> OverloadLiteral<'db> {
```

Comment:
I would a function named `first` expect to return an `Option`, which also seems more in line with what other methods iterating over overloads do. 

In general, the idea of `expect` is to have a message that says why the "expectation holds true". In this case, there isn't any. If returning an `Option` is indeed much more annoying to handle upstream, I suggest to rename the method and add a docstring saying that it's the callers responsibility to ensure this literal has at least one overload